### PR TITLE
fixes #14076 - remove unused Puppetclass hostgroups counter

### PR DESCRIPTION
**Pull request by domcleal**
Replaces #3282.

### CODE REVIEW

File: db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class RemoveHostgroupsCountFromPuppetclasses < ActiveRecord::Migration
+  def up
+    remove_column :puppetclasses, :hostgroups_count
+  end
+
+  def down
+    add_column :puppetclasses, :hostgroups_count, :integer, :default => 0
+    HostgroupClass.all.each do |hgc|
+      Puppetclass.increment_counter(:hostgroups_count, hgc.puppetclass_id)
```
Comment (tbrisker) [inline db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb]: Not sure if it's worth the trouble for a down migration, but we can calculate this entirely in the DB with a single SQL query.

Comment (domcleal) [inline db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb]: A query to group HostgroupClass by puppetclass_id and then update each puppetclass_id's counter?

Comment (tbrisker) [inline db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb]: Something along the lines of:

``` sql
UPDATE puppetclasses SET hostgroups_count = (SELECT count(*) FROM hostgroup_classes WHERE hostgroup_classes.class_id = puppetclasses.id)
```

Technically a nested query, but I think most DBs should be able to optimize it quite well. And without instantiating possibly thousands of objects.

Comment (domcleal) [inline db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb]: Thanks, updated.

**Comment by tbrisker** [comment]
This also needs fixing on this line:
https://github.com/theforeman/foreman/blob/develop/lib/tasks/fix_cached_counters.rake#L13
Other than that, LGTM.

**Comment by domcleal** [comment]
Thanks, fixed the rake task to not update this counter.

**Comment by tbrisker** [comment]
Merged as be6e6efc869f1ea4385cb21bb9dbdb8dc8ea29e2, thanks @domcleal!

**Pull request closed without merging**